src/react-vite/10-nested-routes/nested-routes.md

```diff
@@ -8,45 +8,207 @@
+      element: <App />,
+      children: [
+        {
+          index: true,
+          element: <HomePage />,
+        },
+        {
+          path: 'about',
+          element: <AboutPage />,
+        },
+      ],
+    },
+  ],
+  {
+    basename: import.meta.env.BASE_URL,
+  },
+)
+```
+
+Notice that our home and about pages are under the `children` key. These are actually nested routes of the root path. Route children can accept a `children` property themselves, on and on, until the browser runs out of resources.
+
+The route config for our product page could look like this:
+
+```tsx
+const router = createBrowserRouter(
+  [
+    {
+      path: '/',
+      element: <App />,
+      children: [
+        ...
+        {
+          path: 'product',
+          element: <ProductPage />,
+          children: [
+            {
+              index: "features",
+              element: <Features />,
+            },
+            {
+              index: "where-to-buy",
+              element: <WhereToBuy />,
+            },
+            {
+              index: "support",
+              element: <Support />,
+            },
+            ...
+          ]
+        },
+      ],
+    },
+  ],
+  {
+    basename: import.meta.env.BASE_URL,
+  },
+)
+```
+
+#### Folder structure
+
+Notice that the child page components have simple names like `<Support />` instead of `<ProductSupport />`. Assuming this component is specifically designed for the product page and will not be reused elsewhere, the component file can reside within the product page's folder. Since the files live together, longer names aren't needed and will make your JSX a little harder to read.
+
+The larger directory tree should look like the following:
+
+<div class="directory-list">
+
+- pages
+  - Home
+    - Home.tsx
+    - Home.module.css
+    - Home.test.tsx
+    - index.ts
+  - Product
+    - components
+      - Features
+        - Features.tsx
+        - Features.module.css
+        - Features.test.tsx
+        - index.ts
+      - Support
+        - Support.tsx
+        - Support.module.css
+        - Support.test.tsx
+        - index.ts
+    - Product.tsx
+    - Product.module.css
+    - Product.test.tsx
+    - index.ts
+
+</div>
+
+### Exercise 1
+
+Let's reorganize the existing `/restaurants` routes so they are nested. For this exercise, change the routing config so the restaurant list and details pages are nested under `/restaurants`.
+
+#### Setup
+
+No additional setup needed.
+
+#### Verify
+
+The existing test already cover routing, so no new tests are needed. Ensure the existing tests pass when you ren `npm run test`.
+
+#### Exercise
+
+Refactor the router config in **src/main.tsx** to nest the restaurant routes.
+
+#### Solution 
+
+<details>
+<summary>Click to see the solution</summary>
+
+✏️ Update **src/main.tsx** to be:
+
+@diff ../../../exercises/react-vite/10-nested-routes/01-solution/src/main.tsx ../../../exercises/react-vite/10-nested-routes/01-solution/src/main.tsx only
```

Comment:
I think there’s something wrong with this diff because nothing is highlighted in this section…